### PR TITLE
test: Try once more when RPC connection fails on Windows

### LINKED ISSUE #18548: Tests fail intermittently on Windows (appveyor) "OSError: [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted"
Example:

* https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/31988561#L4009

### PR DESCRIPTION
**Pull request by maflcko**
Fixes: #18548

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Fixed. Tests should now run smoothly again on Linux, Windows and Freebsd.

Freebsd run: https://cirrus-ci.com/build/5888962329837568

File: test/functional/test_framework/authproxy.py [hunk 1]
```diff
@@ -101,23 +101,26 @@ def _request(self, method, path, postdata):
...
             # TODO: Find out why the connection would disconnect occasionally and make it reusable on Windows
+            # Avoid "ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine"
             self._set_conn()
         try:
             self.__conn.request(method, path, postdata, headers)
             return self._get_response()
-        except http.client.BadStatusLine as e:
-            if e.line == "''":  # if connection was closed, try again
+        except (BrokenPipeError, ConnectionResetError):
+            # Python 3.5+ raises BrokenPipeError when the connection was reset
```
Comment (hebasto) [inline test/functional/test_framework/authproxy.py:110]: Is such behavior Windows-specific?

File: test/functional/test_framework/authproxy.py [hunk 2]
```diff
@@ -101,23 +101,26 @@ def _request(self, method, path, postdata):
         if os.name == 'nt':
             # Windows somehow does not like to re-use connections
             # TODO: Find out why the connection would disconnect occasionally and make it reusable on Windows
```
Comment (hebasto) [inline test/functional/test_framework/authproxy.py:103]: Are these lines still relevant?

File: test/functional/test_framework/authproxy.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/test_framework/authproxy.py:110]: Please include the commit you are referring to in the comment. This pull shouldn't change behaviour on non-windows machines in authproxy.

File: test/functional/test_framework/authproxy.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/test_framework/authproxy.py:103]: Yes, why wouldn't they?

**Review by hebasto — APPROVED** [review]
ACK fab98992043f47fa7240d7c1217920d0c4f783a2, I have reviewed the code and it looks OK, I agree it can be merged.

**Pull request merged**